docs/src/components/navigation/table-of-contents.tsx

```diff
@@ -0,0 +1,33 @@
+import slugify from '@sindresorhus/slugify';
+
+type Heading = {
+  level: number;
+  text: string;
+};
+
+type TableOfContentsProps = {
+  headings: Heading[];
+};
+
+export function TableOfContents({ headings }: TableOfContentsProps) {
+  return (
+    <div className="w-[12rem] sticky top-10 lg:top-32 self-start hidden md:block">
+      <h5 className="text-xs uppercase text-stone-500">On this page</h5>
+
+      <ul className="mt-2 flex flex-col gap-3">
+        {headings.map(({ level, text }) => (
+          <li>
+            <a
```

Comment:
Should we use `next/link` here?

Reply:
I went through the same thought process just before committing 😂. Since it is all on-page I don't see any reason that we would need to use Next's link.

Reply:
Yeah I want to remember in EDS docs site, we had some bugginess with the "scroll down to #" when using next/link for the TOC links